### PR TITLE
Feature/waypoints

### PR DESCRIPTION
**Pull request by thomas-gleiforst**
This pull request of the feature/waypoints branch should include everything needed to display a map, add waypoints to the map, store those waypoints in a json file to be recalled later, and overlay a directional compass needle on the map

### CODE REVIEW

File: Electron/RED/ControlCenter.tsx [hunk 1]
```diff
@@ -2,27 +2,58 @@ import React, { Component } from "react"
...
 
 class ControlCenter extends Component<IProps, IState> {
+  waypointsInstance: any
+
   constructor(props: any) {
     super(props)
-    this.state = {}
+    this.state = {
+      storedWaypoints: {},
+      currentCoords: { lat: 0, long: 0 },
```
Comment (EliVerbrugge) [inline Electron/RED/ControlCenter.tsx:31]: Add a button/keybind to center map view back on the rover?

File: Electron/RED/components/GPS.tsx [hunk 2]
```diff
@@ -69,10 +71,13 @@ class GPS extends Component<IProps, IState> {
   }
 
   GPSPosition(data: any) {
+    const currentLat = data[0] / 10000000
```
Comment (EliVerbrugge) [inline Electron/RED/components/GPS.tsx:75]: Comment to describe why this is happening

File: Electron/RED/components/GPS.tsx [hunk 3]
```diff
@@ -50,7 +52,7 @@ class GPS extends Component<IProps, IState> {
       satelliteCount: 255,
       odometer: 0,
       currentLat: 0,
-      currentLon: 0,
+      currentLong: 0,
```
Comment (EliVerbrugge) [inline Electron/RED/components/GPS.tsx]: Lon not long, is consistent across rover

File: Electron/RED/components/Map.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  zoom: number
+  maxZoom: number
+  heading: number
+}
+
+class Map extends Component<IProps, IState> {
+  constructor(props: IProps) {
+    super(props)
+    this.state = {
+      currentLat: 37.951631,
```
Comment (EliVerbrugge) [inline Electron/RED/components/Map.tsx:55]: Leave comment to describe where this is

File: Electron/RED/ControlCenter.tsx (hunk 1 above)
Comment (thomas-gleiforst) [reply Electron/RED/ControlCenter.tsx:31]: This is a good future feature consideration that will be considered later.

**Review by amwgvf — APPROVED** [review]

**Pull request merged**